design-proposals/backend-storage-block-migration.md

```diff
@@ -0,0 +1,67 @@
+# Overview
+This proposal is about supporting the migration of all VMs that use a persistent feature, such as persistent vTPM.
+
+To make this document as simple as possible, let's consider that storage in Kubernetes can either be filesystem (FS) or
+block mode, and either ReadWriteOnce (RWO) or ReadWriteMany (RWX) access mode, with storage classes supporting any
+number of combinations of both.
+
+As of KubeVirt 1.3, Here is how each of the 4 combinations is supported by the backend-storage for persistent VM state:
+- RWO FS: supported by backend-storage but makes VMs non-migratable
+- RWX FS: preferred combination, supported by backend-storage and compatible with VM migrations
+- RWO block: not supported yet, but PRs exist to implement it in a non-migratable way 
+- RWX block: same as above, with migratability functional but considered unsafe, because it leads to 2 pods (source and target virt-launchers) mounting the same ext3 filesystem concurrently.
+
+This is the best we can do with the current implementation, which involves mounting the same partition on migration source and target.
+This proposal is about switching to a different model which would instead create a new backend-storage PVC on migration,
+allowing all 4 cases above to have proper migration support.
+
+## Motivation/Goals
+- Users want to be able to use any storage class for backend-storage
+- Users want all VMs with persistent features to be potentially migratable
+
+## Definition of Users
+Any VM owner that wants to use persistent features, such as TPM or EFI
+
+## User Stories
+As a user, I want to be able to seamlessly enable persistent features while keeping the migratability of my VMs.
+
+## Repos
+kubevirt/kubevirt
+
+# Design
+- When creating a new VM with persistent features, we use either (first match wins):
+  - The storage class specified in the KubeVirt Custom Resource
+  - The storage class marked as the default for virtualization purposes
+  - The storage class marked as the default for the cluster
+- All new persistent storage PVCs will be created with the name `persistent-state-for-<vm_name>-<random_string>` and the new annotation `persistent-state-for-<vm_name>=true`
+- When starting an existing VM with persistent features, we will look for any PVC with the annotation, or any PVC with the legacy name, for backwards compatibility
+- For volume/access modes:
+  - When multiple volume modes are listed in the StorageProfile of the StorageClass, we will default to filesystem (to avoid partitioning complexity)
```

Comment:
I think we can avoid partitioning entirely.  Any StorageClass that supports block should natively support filesystem (like rook-ceph-block)

Reply:
That's huge. Block support is messy and somewhat illogical. Reducing this design to only add RWO FS support would make everything so much simpler and sensible.